mne/forward/forward.py

```diff
@@ -1003,23 +1064,72 @@ def _restrict_gain_matrix(G, info):
 
 
 def compute_depth_prior(G, gain_info, is_fixed_ori, exp=0.8, limit=10.0,
-                        patch_areas=None, limit_depth_chs=False):
-    """Compute weighting for depth prior."""
+                        patch_areas=None, limit_depth_chs=False,
+                        loose_method='svd'):
+    """Compute weighting for depth prior.
+
+    Parameters
+    ----------
+    G : ndarray, shape (n_channels, n_vertices)
+        The gain matrix.
+    gain_info : instance of Info
+        The info associated with the gain matrix.
+    is_fixed_ori : bool
+        Whether or not ``G`` is fixed orientation.
+    exp : float
+        Exponent for the depth weighting.
+    limit : float | None
+        The upper bound on depth weighting. Can be None to be unbounded.
+    patch_areas : ndarray | None
+        Patch areas of the vertices from the forward solution.
+    limit_depth_chs : bool
+        If True, limit depth weighting computation to gradiometers
+        if present (ignoring magnetometers). This can improve the
+        computations on systems that have both (e.g., Neuromag).
+    loose_method : 'svd' | 'sum'
+        When a loose (or free) orientation is used, how the depth weighting
+        for each triplet should be calculated.
+        If 'svd', use the largest singular value of the 3x3 matrix
+        formed by the dot product ``Gk.T @ Gk`` will be used.
+        If 'sum', then ``np.sum(Gk ** 2)`` will be used.
+
+        .. versionadded:: 0.18
+
+    Returns
+    -------
+    depth_prior : ndarray, shape (n_vertices,)
+        The depth prior.
+
+    See Also
+    --------
+    compute_orient_prior
+    """
+    # XXX this perhaps should just take ``forward`` instead of ``G`` and
+    # ``gain_info``. However, it's not easy to do this given that the
+    # mixed norm code requires that ``G`` is whitened before this chunk
```

Comment:
New note added -- if we can get `mixed_norm` not to do the whitening *before* depth weighting, then we can simplify our code a bit, and make this take `forward` just like `compute_orient_prior` does.